yasnippet.el

```diff
@@ -3286,58 +3300,29 @@ If so cleans up the whole snippet up."
 
 ;; Apropos markers-to-points:
 ;;
-;; This was found useful for performance reasons, so that an
-;; excessive number of live markers aren't kept around in the
-;; `buffer-undo-list'. However, in `markers-to-points', the
+;; This was found useful for performance reasons, so that an excessive
+;; number of live markers aren't kept around in the
+;; `buffer-undo-list'.  However, in `markers-to-points', the
 ;; set-to-nil markers can't simply be discarded and replaced with
-;; fresh ones in `points-to-markers'. The original marker that was
-;; just set to nil has to be reused.
+;; fresh ones in `points-to-markers'.  The original marker that was
+;; just set to nil has to be reused (why?).
```

Comment:
Hi. I believe, that, at the time, I did that as a means of preventing a garbage leak. Much later, I believed I realized that there would be no such garbage leak even if the markers aren't `set-buffer`ed to `nil`. 

That is, IIRC it is like the reference from the buffer to the marker is weak. If you don't hold references to the markers anywhere else (but maybe the `buffer-undo-list` does), it should be safe, from a garbage perspective, to just spawn new markers.

At the time I didn't really do that probably because I was lazy or because I was afraid of touching such sensitive code. Have a look at the `snippet-engine` branch which recodes this in a much simpler way. Also, it would be safe to have a look at the return value of `garbage-collect` before and after snippet expansion, particularly the `misc` category, which includes markers.